sbhs/views.py

```diff
@@ -312,39 +329,43 @@ def slot_new(request):
                                         start_time__date=new_slot_date
                                         )
                 if len(new_slot_date_slots) >= settings.LIMIT:
-                    messages.warning(request,'Cannot Book more than {0} \
+                    messages.error(request,'Cannot Book more than {0} \
                         slots in advance in a day'.format(settings.LIMIT))
                 else:
-                    if Slot.objects.check_booked_slots(
-                        new_slot.start_time, all_board_users_id):
-
-                        if new_slot.start_time >= now:
-                            new_slot.start_time = dt.combine(
-                                       new_slot.start_time.date(),
-                                       time(new_slot.start_time.hour,00)
-                                       )
-                            new_slot.end_time = dt.combine(
-                                       new_slot.start_time.date(),
-                                       time(new_slot.start_time.hour,
-                                            settings.SLOT_DURATION
-                                            )
-                                       )
-                            new_slot.user = user
-                            new_slot.save()
-                            messages.success(request,
-                                             'Slot created successfully.'
-                                             )
+                    if new_slot.start_time.time():
```

Comment:
These nested ifs can be cleaned.
For now you can simply check a condition and return a response, this also will be better and more readable.